nodeup/pkg/model/networking/weave.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package networking
+
+import (
+	"k8s.io/kops/nodeup/pkg/model"
+	"k8s.io/kops/upup/pkg/fi"
+)
+
+// WeaveBuilder installs weave
+type WeaveBuilder struct {
+	*model.NodeupModelContext
+}
+
+var _ fi.ModelBuilder = &WeaveBuilder{}
+
+// Build is responsible for configuring the weave
+func (b *WeaveBuilder) Build(c *fi.ModelBuilderContext) error {
+	if b.Cluster.Spec.Networking.Weave == nil {
+		return nil
+	}
+
+	b.AddCNIBinAssets(c, []string{"portmap"})
```

Comment:
It may be that you have to add "loopback" too. This one isn't used by the CNI, but used by kubernetes internally.
If that is the case you should see something like
```
  Warning  FailedCreatePodSandBox  7m24s (x4 over 7m27s)    kubelet, ip-10-140-60-250.eu-central-1.compute.internal  (combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "8e7c8b6aaf6ac0c5a84b8326a724f69b6f21e9d1c5b055e42081b88d70816dd7" network for pod "kube-dns-autoscaler-7647db5967-lwdjw": networkPlugin cni failed to set up pod "kube-dns-autoscaler-7647db5967-lwdjw_kube-system" network: failed to find plugin "loopback" in path [/opt/cni/bin/]
```

Reply:
I didn't notice this while testing, so let's see how tests go in TestGrid.